iothub_client/src/iothubtransport_amqp_cbs_auth.c

```diff
@@ -629,7 +629,9 @@ void authentication_do_work(AUTHENTICATION_HANDLE authentication_handle)
         {
             // Codes_SRS_IOTHUBTRANSPORT_AMQP_AUTH_09_040: [If `instance->state` is AUTHENTICATION_STATE_STARTED and user-provided SAS token was used, authentication_do_work() shall return]
             IOTHUB_CREDENTIAL_TYPE cred_type = IoTHubClient_Auth_Get_Credential_Type(instance->authorization_module);
-            if (cred_type == IOTHUB_CREDENTIAL_TYPE_DEVICE_KEY || cred_type == IOTHUB_CREDENTIAL_TYPE_DEVICE_AUTH)
+            if ( (cred_type == IOTHUB_CREDENTIAL_TYPE_DEVICE_KEY) ||
+                 (cred_type == IOTHUB_CREDENTIAL_TYPE_DEVICE_AUTH) ||
+                 ((cred_type == IOTHUB_CREDENTIAL_TYPE_SAS_TOKEN) && IoTHubClient_Auth_Is_SasToken_Update_Supported(instance->authorization_module)) )
```

Comment:
If I'm getting this right, this would prevent users from passing just one SAS token in the client config and no callback. That would be a breaking change.
I'll post an alternative briefly. Feel free to reply back with details if you believe my understanding is incorrect.

Reply:
I don't think so.
"IoTHubClient_Auth_Is_SasToken_Update_Supported(instance->authorization_module)" will check, if a callback is set.
If not, the "iothubtransport_amqp_cbs_auth.c" will never start any SAS token generation/update or request, but use the SAS token, which is supplied once.